Modules/Relationships/src/Relationships.ConsumerApi/Controllers/RelationshipsController.cs

```diff
@@ -203,7 +203,7 @@ public async Task<IActionResult> CanEstablishRelationship([FromQuery(Name = "pee
         var peerIdentity = await _mediator.Send(new GetIdentityQuery(peerAddress), cancellationToken);
 
         var response = peerIdentity.Status is IdentityStatus.ToBeDeleted
-            ? new CanEstablishRelationshipResponse { CanCreate = false }
+            ? new CanEstablishRelationshipResponse { CanCreate = false, Code = ApplicationErrors.Relationship.PeerIsToBeDeleted().Code }
             : await _mediator.Send(new CanEstablishRelationshipQuery { PeerAddress = peerAddress }, cancellationToken);
```

Comment:
Why is this piece of logic in the controller? I would've moved all this to the handler.

Reply:
I would love to have it there. But the problem is that it uses code from a different module (the Handler is in the Relationships module, but it needs code from the Devices module (the GetIdentityQuery)).

Reply:
Ok, we discussed that in chat extensively, thanks for the insight :)